disnake/client.py

```diff
@@ -432,6 +432,7 @@ def __init__(
         self._first_connect: asyncio.Event = asyncio.Event()
         self._connection._get_websocket = self._get_websocket
         self._connection._get_client = lambda: self
+        self._token: str | None = None
```

Comment:
Since we still support python <= 3.10, we can't use this hint or it will raise an error
`self._token: Optional[str]` would be more appropriate

Reply:
We actually can as long as there is a `from __future__ import annotations` at the top, but i agree, albeit for a different reason: it would look inconsistent with the rest of the codebase